streaming/base/dataset.py

```diff
@@ -384,17 +439,26 @@ def _download_file(self, basename: str) -> str:
         Returns:
             str: Local cache filename.
         """
+        # Calculate paths.
         if self.remote is None:
             remote = None
         else:
             remote = os.path.join(self.remote, self.split, basename)
         local = os.path.join(self.local, self.split, basename)
+
+        # Attempt to download, possibly repeating on faiure.
         for _ in range(1 + self.download_retry):
             try:
                 download(remote, local, self.download_timeout)
-            except:
+            except Exception as e:
+                print('Download exception:', e)
```

Comment:
Let's use logging instead of print

```
import logging
log = logging.getLogger(__name__)
log.exception("message")
```

Reply:
Just removed the debug print for now.

I feel like we want to come up with something more comprehensive about what we track and log across workers, ranks, and nodes...